game.sy

```diff
@@ -256,65 +268,74 @@ collision_check :: fn delta:float {
     arena.update' delta
 }
 
-update_menu :: fn delta: float -> void {
-    main_menu.update' delta
-    if main_menu.line_selected {
-        if main_menu_lines[main_menu.line_index] == "Connect" {
-            // When we break out of the client loop, return here.
-            reset_state'
-            if client.connect("127.0.0.1") {
-                show_main_menu = false
-                client.main_loop'
-                show_main_menu = true
+menu_state :: GameState{
+    update: fn delta: float -> void {
+        main_menu.update' delta
+        if main_menu.line_selected {
+            if main_menu_lines[main_menu.line_index] == "Connect" {
+                // When we break out of the client loop, return here.
+                reset_state'
+                if client.connect("127.0.0.1") {
+                    current_game_state = hockey_state
+                    client.main_loop'
+                    current_game_state = menu_state
+                    main_menu.line_index = 0
+                }
+                //TODO(gu): else show connection error in ui in some way
+            } else if main_menu_lines[main_menu.line_index] == "Start server" {
+                reset_state'
+                current_game_state = hockey_state // ?
```

Comment:
Make a function for swapping states? So we can do stuff like clean up? Or do we need to do that ever? Do we just make a new gamestate?

Reply:
Not sure if we actually need that right now, if there are more complex states maybe. It could definitely be done.